### PR TITLE
Write contract code in raw

### PR DESCRIPTION
**Pull request by freesig**
replaces #796

### CODE REVIEW

**Comment by Voxelot** [comment]
Don't see how these changes are related to the fuel-asm change. We haven't landed & released the prs for avoiding code cloning yet so this doesn't seem like a blocker or replacement for the fuel-asm upgrade PR.

**Comment by freesig** [comment]
> Don't see how these changes are related to the fuel-asm change. We haven't landed & released the prs for avoiding code cloning yet so this doesn't seem like a blocker or replacement for the fuel-asm upgrade PR.

I must have used the auto generated title. Yeh this has nothing to do with ASM it's just based on those changes because the related pr in fuel VM already has the ASM changes

**Review by xgreenx — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]

File: crates/fuel-core/src/state/rocks_db.rs [hunk 1]
```diff
@@ -168,15 +168,19 @@ impl KeyValueStore for RocksDb {
...
         let prev = self.get(key, column)?;
+        // FIXME: This is a race condition. We should use a transaction.
         self.db
             .put_cf(&self.cf(column), key, value)
             .map_err(|e| DatabaseError::Other(e.into()))
             .map(|_| prev)
     }
 
     fn delete(&self, key: &[u8], column: Column) -> DatabaseResult<Option<Vec<u8>>> {
+        // FIXME: This is a race condition. We should use a transaction.
```
Comment (Dentosal) [inline crates/fuel-core/src/state/rocks_db.rs:181]: This should have a github issue. I'm not sure how critical these are, but typically fixing a race condition should ba a high priority.

Comment (xgreenx) [reply crates/fuel-core/src/state/rocks_db.rs:181]: It is also related to the todo above

**Pull request merged**